chapters/ch47.md

```diff
@@ -42,17 +42,17 @@ Chúng ta có thể hình dung "pipeline" của hai thành phần này như sau:
 
 > There has been a recent trend toward replacing pipeline systems with a single learning algorithm. An **end-to-end learning algorithm** for this task would simply take as input the raw, original text "This is a great mop!", and try to directly recognize the sentiment:
 
-Xu hướng gần đây là thay đổi hệ thống pipeline với một thuật toán duy nhất. Một **thuật toán đầu-cuối** cho tác vụ này sẽ đơn giản là thu thập văn bản gốc "Cây lau nhà này thật tuyệt!", và cố gắng trực tiếp nhận ra cảm xúc từ nó:
+Xu hướng gần đây là thay đổi hệ thống pipeline bởi một thuật toán duy nhất. Một **thuật toán đầu-cuối** cho tác vụ này chỉ cần thu về văn bản gốc "Cây lau nhà này thật tuyệt!", rồi cố gắng trực tiếp nhận ra cảm xúc từ văn bản gốc đó:
 
 ![img](../imgs/C47_02.png)
 
 > Neural networks are commonly used in end-to-end learning systems. The term "end-to-end" refers to the fact that we are asking the learning algorithm to go directly from the input to the desired output. I.e., the learning algorithm directly connects the "input end" of the system to the "output end."
 
-Mạng neural được sử dụng phổ biến trong các hệ thống đầu-cuối. Thuật ngữ "đầu-cuối" phản ánh thực tế là chúng ta yêu cầu thuật toán chạy trực tiếp từ đầu vào cho đến đầu ra mong muốn. Tức là, thuật toán học kết nối trực tiếp "đầu vào" cho đến "đầu ra" của hệ thống.
+Mạng nơ-ron được sử dụng phổ biến trong các hệ thống đầu-cuối. Thuật ngữ "đầu-cuối" phản ánh việc chúng ta yêu cầu thuật toán chạy trực tiếp từ đầu vào cho đến đầu ra mong muốn: Thuật toán học nối trực tiếp "đầu vào" và "đầu ra" của hệ thống.
```

Comment:
Chỗ này ông Andrew viết nhấn mạnh "input end" và "output end" để giải thích từ "end-to-end". 
Mình dịch là "đầu-cuối" thì có cách nào viết lại câu này cho có chữ "đầu và cuối" luôn không nhỉ? Như hiện tại "đầu vào - đầu ra" chưa giải thích rõ lắm ý của thuật ngữ "đầu-cuối", mà mình chưa nghĩ ra cách dịch nào hay hơn.

Reply:
đoạn này mình nghĩ nên giải thích hẳn nghĩa của từ end-to-end ra. Bác ấy giải thích cũng không rõ ý lắm. Vì end-to-end learning là tất cả các quá trình từ đầu đến cuối đều được học cùng nhau.

Reply:
Dịch thô là "phía đầu" và "phía cuối" chăng?
> Thuật toán học nối trực tiếp "phía đầu" và "phía cuối" của hệ thống